src/main/java/io/syndesis/verifier/v1/metadata/SalesforceMetadataAdapter.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.verifier.v1.metadata;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchema;
+import com.fasterxml.jackson.module.jsonSchema.types.ObjectSchema;
+import com.fasterxml.jackson.module.jsonSchema.types.SimpleTypeSchema;
+
+import org.apache.camel.component.extension.MetaDataExtension.MetaData;
+import org.springframework.stereotype.Component;
+
+@Component("salesforce-adapter")
+public final class SalesforceMetadataAdapter implements MetadataAdapter {
```

Comment:
Some test to the adapter could also help in clarifying its usage and what kind of data is transformed into which format.

Reply:
I've added a test of the `SalesforceMetadataAdapter`

Reply:
thanks !